bech32m.py

```diff
@@ -0,0 +1,341 @@
+
+
+def bech32_verify_checksum(hrp, data):
+    """Verify a checksum given HRP and converted data characters."""
+    const = bech32_polymod(bech32_hrp_expand(hrp) + data)
+    if const == BECH32M_CONST:
+        return 1
+    else:
+        return None
+
+
+def bech32m_create_checksum(hrp, data):
+    """Compute the checksum values given HRP and data."""
+
+    values = bech32_hrp_expand(hrp) + data
+    polymod = bech32_polymod(values + [0, 0, 0, 0, 0, 0]) ^ BECH32M_CONST
+    return [(polymod >> 5 * (5 - i)) & 31 for i in range(6)]
+
+
+def convertbits(data, frombits, tobits, pad=True):
+    """General power-of-2 base conversion."""
+    acc = 0
+    bits = 0
+    ret = []
+    maxv = (1 << tobits) - 1
+    max_acc = (1 << (frombits + tobits - 1)) - 1
+
+    for value in data:
+        if value < 0 or (value >> frombits):
+            return None
+        acc = ((acc << frombits) | value) & max_acc
+        bits += frombits
+        while bits >= tobits:
+            bits -= tobits
+            ret.append((acc >> bits) & maxv)
+    if pad:
+        if bits:
+            ret.append((acc << (tobits - bits)) & maxv)
+    elif bits >= frombits or ((acc << (tobits - bits)) & maxv):
+        return None
+    return ret
+
+
+def check_if_str_is_base64(check_str):
+    """check if given string is valid base64 string"""
+
+    base64_charset = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/="
+    if check_str == "":
+        return True
+
+    elif type(check_str) is not str:
+        print("Input given not a string. If input is raw bytes, select -bin as input format")
+        return False
+
+    elif len(check_str) % 4 != 0:
+        print("Invalid length of base 64 string. Check input string and input format")
+        return False
+
+    elif not all(x in base64_charset for x in check_str):
+        print("Non base64 Character Entered in String")
+        return False
+
+    else:
+        return True
+
+
+def check_if_str_is_hex(check_str):
+    """
+    check if given string is valid hexadecimal string
+    :param check_str: string to be validated for hexadecimal encoding
+    :return: true if string has valid hexadecimal encoding
+    """
+
+    hex_charset = "ABCDEFabcdef0123456789"
+    if check_str == "":
+        return True
+
+    elif type(check_str) is not str:
+        print("Input given not a string. If input is raw bytes, select \"bin\" as input format")
+        return False
+
+    elif len(check_str) % 2 != 0:
+        print("Invalid Length of Hex String")
+        return False
+
+    elif not all(x in hex_charset for x in check_str):
+        print("Non Hex Character Entered in String")
+        return False
+
+    else:
+        return True
+
+
+def check_if_str_is_bin(check_str):
+    """
+    check if input string has raw bytes
+    :param check_str: string to be checked
+    :return: true if string is raw bytes else false
+    """
+    try:
+        b = (check_str.decode()).encode()
```

Comment:
issue (blocking): If a string contains a non utf-8 character, the `decode()` crashes.